### PR TITLE
feat: run workflow with option for multiple executions, improve cypress execution logic

### PR DESCRIPTION
**Pull request by johnrwatson**
Adds the possibility of driving cypress for test magnitude using `VITE_SI_CYPRESS_MULTIPLIER` to multiply the amount of times the tests are executed on each run to simulate user load. Additionally modified the CI build to be a matrix (of 1) so we can easily multiply on that vector too to give larger scale if needed. i.e.

CI Executor Multiples x VITE_SI_CYPRESS_MULTIPLIER x Test Number = Load

I also added a Posthog command to push a custom UUID into Posthog for each test to allow us to easily look at the user session for each event. This should be super helpful when we get more complicated tests or fragile tests to figure out what is wrong.

### CODE REVIEW

File: .github/workflows/e2e-validation.yml [hunk 1]
```diff
@@ -4,14 +4,19 @@ on:
   workflow_dispatch:
   push:
     branches:
-      - main
+      - feat/add-more-synthetics
```
Comment (stack72) [inline .github/workflows/e2e-validation.yml:7]: I don't think we want this right @johnrwatson ?

Comment (stack72) [reply .github/workflows/e2e-validation.yml:7]: Also, ANY push to main will trigger these tests - that doesn't feel right as that code won't have made production yet - should this be limited to any changes to the cypress tests?

**Comment by johnrwatson** [comment]
Looks good:
image

**Comment by johnrwatson** [comment]
bors merge

**Pull request merged**